lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -706,3 +752,44 @@ Object? domCallConstructorString(String constructorName, List<Object?> args) {
 
 bool domInstanceOfString(Object? element, String objectType) =>
     js_util.instanceof(element, domGetConstructor(objectType)!);
+
+@JS('Object.getPrototypeOf')
+external Object? domObjectGetPrototypeOf(Object? value);
+
+@JS('Object.prototype')
+external Object? get domObjectPrototype;
+
+@JS('Object.keys')
+external List<Object?> domObjectKeys(Object? object);
+
+bool isJavaScriptArray(Object? value) => domInstanceOfString(value, 'Array');
+bool isJavaScriptSimpleObject(Object? value) {
+  final Object? proto = domObjectGetPrototypeOf(value);
+  return proto == null || proto == domObjectPrototype;
+}
+
+// This will likely be in js_util before this CL lands.
```

Comment:
I was going to ask, shouldn't this be in `js_util` for symmetry with `jsify`?

Reply:
Yes, good catch. I upstreamed this change and was just waiting for it to roll into Flutter. I've updated the CL to use dartify from `js_util`. I think its slightly more generic, so hopefully it will work for our use cases here, but otherwise we can specialize it again.